src/api/cli/integration/integration_test.go

```diff
@@ -1341,3 +1339,311 @@ func TestLiveWalletDir(t *testing.T) {
+		loop:
+			for {
+				select {
+				case <-time.After(30 * time.Second):
+					t.Fatal("Wait tx confirmation timeout")
+				case <-tk.C:
+					if isTxConfirmed(t, txid) {
+						break loop
+					}
+				}
+			}
+
+			tc.checkTx(t, txid)
+		})
+	}
+
+	// Send with too small decimal value
+	errMsg := []byte("ERROR: invalid amount, too many decimal places. See 'skycoin-cli send --help'")
+	for i := 0.0001; i <= 0.0009; i++ {
+		v, err := droplet.ToString(uint64(i * droplet.Multiplier))
+		require.NoError(t, err)
+		name := fmt.Sprintf("send %v", v)
+		t.Run(name, func(t *testing.T) {
+			output, err := exec.Command(binaryPath, "send", w.Entries[0].Address.String(), v).CombinedOutput()
+			require.NoError(t, err)
+			output = bytes.Trim(output, "\n")
+			require.Equal(t, errMsg, output)
+		})
+	}
+}
+
+func getTransaction(t *testing.T, txid string) *webrpc.TxnResult {
+	output, err := exec.Command(binaryPath, "transaction", txid).CombinedOutput()
+	require.NoError(t, err)
+
+	var tx webrpc.TxnResult
+	err = json.NewDecoder(bytes.NewReader(output)).Decode(&tx)
+	require.NoError(t, err)
+
+	return &tx
+}
+
+func isTxConfirmed(t *testing.T, txid string) bool {
+	tx := getTransaction(t, txid)
+	return tx.Transaction.Status.Confirmed
+}
+
+// checkCoinhours checks if the address coinhours in transaction are correct
+func checkCoinsAndCoinhours(t *testing.T, tx *webrpc.TxnResult, addr string, coins, coinhours uint64) {
+	addrCoinhoursMap := make(map[string][]visor.ReadableTransactionOutput)
+	for _, o := range tx.Transaction.Transaction.Out {
+		addrCoinhoursMap[o.Address] = append(addrCoinhoursMap[o.Address], o)
+	}
+
+	os, ok := addrCoinhoursMap[addr]
+	if !ok {
+		t.Fatalf("transaction doesn't have receiver of address: %v", addr)
+	}
+
+	var totalCoins, totalHours uint64
+	for _, o := range os {
+		c, err := droplet.FromString(o.Coins)
+		if err != nil {
+			t.Fatalf("%v", err)
+		}
+		totalCoins += c
+		totalHours += o.Hours
+	}
+
+	require.Equal(t, coins, totalCoins)
+	require.Equal(t, coinhours, totalHours)
+}
+
+// prepareAndCheckWallet prepares wallet for live testing.
+// Returns *wallet.Wallet, total coin, total hours.
+// Confirms that the wallet meets the minimal requirements of coins and coinhours.
+func prepareAndCheckWallet(t *testing.T, miniCoins, miniCoinHours uint64) (*wallet.Wallet, uint64, uint64) {
+	walletDir, walletName := getWalletPathFromEnv(t)
+	walletPath := filepath.Join(walletDir, walletName)
+	// Checks if the wallet does exist
+	if _, err := os.Stat(walletPath); os.IsNotExist(err) {
+		t.Fatalf("Wallet file: %v does not exist", walletPath)
+	}
+
+	// Loads the wallet
+	w, err := wallet.Load(walletPath)
+	if err != nil {
+		t.Fatalf("Load wallet failed: %v", err)
+	}
+
+	if len(w.Entries) < 3 {
+		// Generates addresses
+		_, err = w.GenerateAddresses(uint64(3 - len(w.Entries)))
+		if err != nil {
+			t.Fatalf("Wallet generateAddress failed: %v", err)
+		}
+	}
+
+	outputs := getWalletOutputs(t, walletPath)
```

Comment:
we should move this above `len(w.Entries)` check

Reply:
No, if the wallet doesn't have any address entries, we won't get any outputs from the wallet. We do `len(w.Entries)` check first, and if the wallet has less than 3 address, we will generate more addresses.

Reply:
The test checks if the wallet has no coins, if the wallet has no coins creating more addresses will not generate coins.